tests/framework/integration/settable_logger.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"io"
+	"sync"
+
+	"google.golang.org/grpc/grpclog"
+)
+
+// settableLoggerV2 is thread-safe.
+// Copied from https://github.com/grpc-ecosystem/go-grpc-middleware/tree/v1.4.0/logging/settable.
```

Comment:
We have already upgraded to `grpc-ecosystem/go-grpc-middleware/v2` in this PR, I am a little concerned why we still need to copy code from `grpc-ecosystem/go-grpc-middleware v1` ? Is it possible to do it in consistent & v2 style?

Also link to the original PR which introduced the `var grpc_logger grpc_logsettable.SettableLoggerV2` in integration test. https://github.com/etcd-io/etcd/pull/12781

Reply:
[logsettable package](https://github.com/grpc-ecosystem/go-grpc-middleware/tree/v1.4.0/logging/settable) was removed in v2 so I opted to copy from v1 in lack of alternative.